### PR TITLE
Remove async profile creation

### PR DESCRIPTION
**Pull request by konradko**
https://uktrade.atlassian.net/browse/ED-1440

Goodnight, sweet prince.

### CODE REVIEW

File: enrolment/views.py [hunk 1]
```diff
@@ -18,20 +13,13 @@ class EnrolmentCreateAPIView(CreateAPIView):
     http_method_names = ("post", )
 
     def create(self, request, *args, **kwargs):
```
Comment (richtier) [inline enrolment/views.py]: now this view is simply calling `serializer.save`, we can remove the `create` method, and the default behaviour of `CreateAPIView` will do the leg work for us.

File: supplier/signals.py [hunk 2]
```diff
@@ -1,22 +0,0 @@
...
-
-
-def send_confirmation_email(sender, instance, created, *args, **kwargs):
-    sync_creation = settings.FEATURE_SYNCHRONOUS_PROFILE_CREATION
-    if not created or not instance.company_email or sync_creation:
-        return
-
-    context = {'url': settings.COMPANY_EMAIL_CONFIRMATION_URL}
-    text_body = render_to_string('confirmation_email.txt', context)
-    html_body = render_to_string('confirmation_email.html', context)
```
Comment (richtier) [inline supplier/signals.py]: we can now delete `confirmation_email.html` and `confirmation_email.txt`.

**Comment by richtier** [comment]
really nice refactor :)

File: enrolment/views.py (hunk 1 above)
Comment (konradko) [inline enrolment/views.py]: Keeping this as we do not send `data` in the request - the structure expected by the serializer is created here. Will create a separate task for this

**Pull request merged**